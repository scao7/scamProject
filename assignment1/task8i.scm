
(define (ecfi height)
	(define e 2.0)
	;(define fact 0.0)
	(define n 2)
	(define i 0)
	(define (iter fact i)
		fact
		(if(< i height)
		(iter (/ 1 (+ ( / 1 (+ (/ 1 (+ 1 fact)) (* (- height i) 2.0))) 1)) (+ i 1))
		(+ e fact)
		)
	)
	(iter 0.0 0)
	
)

(define (main)
	(setPort (open (getElement ScamArgs 1) 'read))
	(define arg (readExpr))
	(println "(ecfi " arg ") is " (fmt "%.25f" (ecfi arg)))
	
	;change for test case 
) 
