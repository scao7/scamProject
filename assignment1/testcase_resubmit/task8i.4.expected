(ecfi 29) is 2.7182818284590455348848081
